### PR TITLE
feat(Filename): add API function to extract filename (with ou with extension) from a path

### PR DESCRIPTION
**Pull request by panquez**

### CODE REVIEW

File: include/geode/basic/filename.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+#pragma once
+
+#include <geode/basic/common.h>
+
+namespace geode
+{
+    std::string opengeode_basic_api filename_from_path(
```
Comment (BotellaA) [inline include/geode/basic/filename.h:30]: I do not like function which can do several things. One function = One contract. Could you provide two functions? One for each contract?

What do you think?

Comment (panquez) [reply include/geode/basic/filename.h:30]: I have no opinion on that question. I can do two functions. The major point for me is the return type. I would prefer absl::string_view but I do not see how to do it using ghc::filesystem::path

Comment (BotellaA) [reply include/geode/basic/filename.h:30]: you cannot with filesystem, return type should own the data.

Comment (panquez) [reply include/geode/basic/filename.h:30]: Ok. So std::string is good enough ?

Comment (BotellaA) [reply include/geode/basic/filename.h:30]: yes.. until we move to C++17 for good

**Pull request merged**